rclcpp/src/rclcpp/duration.cpp

```diff
@@ -214,6 +214,13 @@ Duration::nanoseconds() const
   return rcl_duration_.nanoseconds;
 }
 
+Duration
+Duration::DURATION_MAX()
+{
+  Duration duration_max(std::numeric_limits<int32_t>::max(), 999999999);
```

Comment:
@tfoote This PR and #524 say a duration is limited by the max value that can be given to the constructor accepting seconds and nanoseconds, but there is [another constructor taking nanoseconds directly](https://github.com/ros2/rclcpp/blob/25a9b4e339867bacb8ba5b92b955420cec86596a/rclcpp/src/rclcpp/duration.cpp#L45), and that seems to accept a number of nanoseconds as large as the [64bit `rcutils` type](https://github.com/ros2/rcutils/blob/4cbd05d9c09886754f41747afa705f15d02c6818/include/rcutils/time.h#L46). Should this instead be `std::numeric_limits<rcl_duration_value_t>::max()`? Alternatively, should the nanoseconds constructor restrict the maximum size to int32_t seconds + 999999999 nanoseconds?

Reply:
I think that the nanoseconds constructor should limit it to this representation. Otherwise we'll overflow the message datastructure.